docs/DeveloperGuide.md

```diff
@@ -797,9 +797,167 @@ Use case ends.
+    <br>
+    Use case ends.
+* 2d. Student is already in lesson.
+  * 2d1. TAHub shows an error message stating that the Student is already in the lesson.
+    <br>
+    Use case ends.
+
+
+**<u>Use case: UC18 - Remove Student from Lesson</u>**
+
+**Precondition:** The Lesson exists in TAHub.
+
+**MSS:**
+1. User requests to remove a specific student from the lesson by providing the necessary details (Lesson Index, Student Name)
+2. TAHub validates the inputs.
+3. TAHub removes the student from the lesson.
+4. TAHub displays the updated lesson list.
+   <br>
+   Use case ends.
+
+**Extensions:**
+* 2a. Invalid Lesson Index.
+  * 2a1. TAHub shows an error message stating that the Lesson Index is invalid.
+    <br>
+    Use case ends.
+* 2b. Student Name not Found in Student List.
+  * 2b1. TAHub shows an error message stating that the Student does not exist.
+    <br>
+    Use case ends.
+* 2c. Student is not in lesson
+  * 2c1. TAHub shows an error message stating that the Student is not in the lesson.
+    <br>
+    Use case ends.
+
+
+**<u>Use case: UC19 - Delete Lesson</u>**
+
+**Precondition:** The Lesson exists in TAHub.
+
+**MSS:**
+1. User requests to delete a specific lesson by providing the necessary details (Lesson Index).
+2. TAHub verifies the Lesson Index.
+3. TAHub deletes the lesson at the index in the Lesson List.
+   <br>
+   Use case ends.
+
+**Extensions:**
+* 2a. Invalid Lesson Index
+  * 2a1. TAHub shows an error message stating that the Lesson Index is invalid.
+    <br>
+    Use case ends.
+
+
+**<u>Use case: UC20 - Mark Student Attendance in Lesson</u>**
+
+**Precondition:** The Student & Lesson exists in TAHub. Student is already in the Lesson.
+
+**MSS:**
+1. User requests to mark a specific student's attendance in the lesson by providing the necessary details (Lesson Index, Student Name, Attendance)
+2. TAHub validates the inputs.
+3. TAHub marks the student's attendance in the lesson.
+4. TAHub displays the updated student attendance.
+   <br>
+   Use case ends.
+
+**Extensions:**
+* 2a. Invalid Lesson Index.
+  * 2a1. TAHub shows an error message stating that the Lesson Index is invalid.
+    <br>
+    Use case ends.
+* 2b. Student is not in lesson.
+  * 2b1. TAHub shows an error message stating that the Student is not in the lesson.
+    <br>
+    Use case ends.
+* 2c. Invalid Attendance.
+  * 2c1. TAHub shows an error message stating that the Attendance is invalid.
+    <br>
+    Use case ends.
+
+
+**<u>Use case: UC21 - Mark Student Participation in Lesson</u>**
+
+**Precondition:** The Student & Lesson exists in TAHub. Student is already in the Lesson.
+
+**MSS:**
+1. User requests to mark a specific student's participation in the lesson by providing the necessary details (Lesson Index, Student Name, Participation)
+2. TAHub validates the inputs.
+3. TAHub marks the student's participation in the lesson.
+4. TAHub displays the updated student attendance.
+   <br>
+   Use case ends.
```

Comment:
Also, you may want to add as an extension that if the participation is set to >0, the student's attendance is also set to true.

Reply:
Will Do!